mitiq/pec/representations/optimal.py

```diff
@@ -124,9 +125,15 @@ def find_optimal_representation(
     Returns: The optimal OperationRepresentation.
     """
     ideal_cirq_circuit, _ = convert_to_mitiq(ideal_operation)
-    ideal_matrix = kraus_to_super(
-        cast(List[npt.NDArray[np.complex64]], kraus(ideal_cirq_circuit))
-    )
+
+    ops = list(ideal_cirq_circuit.all_operations())
+    super_ops = [
+        kraus_to_super(cast(List[npt.NDArray[np.complex64]], kraus(op)))
+        for op in ops
+    ]
```

Comment:
Two suggestions put into one:

1. No need to cast `all_operations()` to a list since we are just going to iterate over it.
2. `super_ops` does not need to be a list either since we are going to iterate over it.
```suggestion
    super_ops = (
        kraus_to_super(cast(List[npt.NDArray[np.complex64]], kraus(op)))
        for op in ideal_cirq_circuit.all_operations()
    )
```

Reply:
Oh I also just found [`cirq.operation_to_superoperator`](https://quantumai.google/reference/python/cirq/operation_to_superoperator). It looks like under the hood it does all the Kraus stuff for us.

Reply:
Nice! I'll try that next week and maybe add some tests with a reset gate. I don't think this PR is essential for the release and we can proably keep it out. But if you want, you can merge it in the current form and I'll make the updates in a second step.